src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -256,6 +257,12 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
     //executorName is to distinguish different types of nodes
     //taskType is the type of task in mesos to launch on the node
     //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
```

Comment:
Talked to @abhay-agarwal. These changes would not have an effect on this bug so I am not okay with adding more complexity to the code. I think we should revert these changes and a TODO to clean up the check the nodes and return false within this method. Ideally, we should never be calling launchTask if we don't want to launch a task. I'm also okay with doing the clean up now to check to see that tasks are waiting to launch being entering the launchTask method.